packages/catalog-search/src/SearchBox.jsx

```diff
@@ -54,40 +59,61 @@ export const SearchBoxBase = ({
     }
   };
 
+  const handleSuggestionSubmit = (query, hit) => {
+    // If an override is provided, call it with the hit
+    if (suggestionSubmitOverride !== undefined) {
+      suggestionSubmitOverride(hit);
+    }
+
+    // When a suggested search is clicked, set the course tile to the search query
+    setSearchQuery(query);
+    // Do all the regular submit stuff
+    handleSubmit(query);
+  };
+
   useEffect(() => {
     document.addEventListener('click', () => setShowSuggestions(false));
   });
 
+  // Track the focused element
+  const focusedElement = useActiveElement();
+
   useEffect(() => {
     let isMounted = true;
-    const fetchCourses = setTimeout(async () => {
-      if (searchQuery !== '' && isMounted) {
-        const { hits, nbHits } = await index.search(searchQuery, {
-          filters,
-          attributesToHighlight: ['title'],
-          attributesToRetrieve: [
-            'key',
-            'content_type',
-            'title',
-            'authoring_organizations',
-            'aggregation_key',
-            '_highlightResult',
-            'program_type',
-          ],
-        });
-        if (nbHits > 0 && isMounted) {
-          setAutocompleteHits(hits);
-          setShowSuggestions(true);
+
+    // If the component is provided an index and the current focused element is the input field,
+    // start the suggestion request to algolia
+    if (index !== undefined && focusedElement.classList.contains(SEARCH_BOX_CLASS_NAME)) {
+      const fetchCourses = setTimeout(async () => {
```

Comment:
[clarification] (probably a separate task) It might not make sense to use `setTimeout` here. Would this potentially be making many more queries to Algolia than we need to be making, in that on every keystroke, this `useEffect` will create a new timeout function that resolves after a second? Typically for autocomplete/typeahead, `debounce` would be used where the callback function doesn't get called until the function stops being called for a specified amount of time (e.g., don't query Algolia until 100ms _after_ the last character I type).

Reply:
Agreed, we should check how this operates. I think Alex is updating existing code in place but I also agree thatt debounce seems like a better pattern when doing autocomplete, in general, that is commonly done.

I am also open to the idea of doing that as a separate task if Alex think it's cleaner, but yes totally on board

Reply:
e.g., https://github.com/edx/frontend-app-admin-portal/blob/f385ac0795e264f6c5c42b7d413ea3e5d2e368eb/src/components/subscriptions/licenses/LicenseManagementTable/index.jsx

Reply:
In case it's helpful, here's some prior art on using `debounce` from `lodash.debounce`: https://github.com/openedx/frontend-app-admin-portal/blob/869afb279807f0cd1cf955314a2bcbe48d3ceda5/src/components/SubsidyRequestManagementTable/data/hooks.js#L100

Reply:
Oh, beat me to it @binodpant haha! :D Now there's 2 examples.